app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/EditCoInsuredDestination.kt

```diff
@@ -2,37 +2,160 @@ package com.hedvig.android.feature.editcoinsured.ui
 
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Divider
+import androidx.compose.material3.Surface
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
 import com.hedvig.android.core.ui.appbar.m3.TopAppBarWithBack
+import com.hedvig.android.core.ui.rememberHedvigDateTimeFormatter
+import com.hedvig.android.feature.editcoinsured.data.CoInsured
+import com.hedvig.android.feature.editcoinsured.data.Member
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.datetime.LocalDate
 import org.koin.androidx.compose.koinViewModel
 import org.koin.core.parameter.parametersOf
 
 @Composable
 internal fun EditCoInsuredDestination(
   contractId: String,
+  allowEdit: Boolean,
   navigateUp: () -> Unit,
 ) {
   val viewModel: EditCoInsuredViewModel = koinViewModel { parametersOf(contractId) }
   val uiState by viewModel.uiState.collectAsStateWithLifecycle()
 
   EditCoInsuredScreen(
     navigateUp,
+    allowEdit,
     uiState,
   )
 }
 
 @Composable
 private fun EditCoInsuredScreen(
   navigateUp: () -> Unit,
+  allowEdit: Boolean,
   uiState: EditCoInsuredState,
 ) {
   Column(Modifier.fillMaxSize()) {
     TopAppBarWithBack(
-      title = "Insured people", // TODO
+      title = stringResource(id = hedvig.resources.R.string.COINSURED_EDIT_TITLE),
       onClick = navigateUp,
     )
+    CoInsuredList(uiState, allowEdit)
+  }
+}
+
+@Composable
+private fun CoInsuredList(uiState: EditCoInsuredState, allowEdit: Boolean) {
+  val dateTimeFormatter = rememberHedvigDateTimeFormatter()
+  Column {
+    uiState.member?.let {
+      InsuredRow(
+        displayName = it.displayName,
+        details = it.ssn,
+        hasMissingInfo = false,
+        allowEdit = false,
+        isMember = true,
+        onRemove = { },
+        onEdit = { },
+      )
+    }
+    Divider()
+    uiState.coInsured.forEachIndexed { index, coInsured ->
+      InsuredRow(
+        displayName = coInsured.displayName,
+        details = coInsured.details(dateTimeFormatter),
+        hasMissingInfo = coInsured.hasMissingInfo,
+        isMember = false,
+        allowEdit = allowEdit,
+        onRemove = { },
+        onEdit = { },
+      )
+      if (index < uiState.coInsured.size - 1) {
+        Divider()
+      }
```

Comment:
I've realized lately that just doing it the other way around, putting it as the first thing in the loop, with a check on the first index is definitely an easier approach not to make any mistakes with looking for the wrong index or something like that. Now doing these `size - 1` etc checks are always prone to a programmer mistake I think.

Here https://github.com/HedvigInsurance/android/blob/03cecb9b5c957d0f15084f67cbe0e49d7e102d7d/app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments/history/PaymentHistoryDestination.kt#L129-L131 is an example of this, it's much harder to make a mistake on checking if you are simply not on the first item instead.